public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -14,50 +23,137 @@ interface BaseAlertmanagerArgs {
 }
 
 export interface NotificationTemplate {
+  uid: string;
   name: string;
   template: string;
   provenance: string;
 }
 export function useNotificationTemplates({ alertmanager }: BaseAlertmanagerArgs) {
   const { useGetAlertmanagerConfigurationQuery } = alertmanagerApi;
+  const { useListNamespacedTemplateGroupQuery } = templatesApi;
+
+  const k8sApiSupported = shouldUseK8sApi(alertmanager);
+
+  const k8sTemplatesRequestState = useListNamespacedTemplateGroupQuery(
+    { namespace: getK8sNamespace() },
+    {
+      skip: !k8sApiSupported || alertmanager !== GRAFANA_RULES_SOURCE_NAME,
```

Comment:
Do we need the second check here? 🤔 If the k8s API isn't supported, then it won't be Grafana anyway (I know that slightly ties to the inner workings of the `shouldUseK8sApi` method, but still, WDYT?)